src/test/shell/bazel/bazel_workspaces_test.sh

```diff
@@ -292,6 +292,24 @@ function test_file() {
   ensure_contains_exactly 'executable: true' 1
 }
 
+function test_read() {
```

Comment:
Add a test with non-Latin-1 UTF-8 characters. Add a test for legacy_utf8=False

Reply:
Done and done.